x-pack/plugins/apm/server/lib/helpers/transactions/index.ts

```diff
@@ -108,11 +108,10 @@ export function getDurationFieldForTransactions(
     type = typeOrSearchAgggregatedTransactions;
   }
 
-  if (type === ApmDocumentType.ServiceTransactionMetric) {
-    return TRANSACTION_DURATION_SUMMARY;
-  }
-
-  if (type === ApmDocumentType.TransactionMetric) {
+  if (
+    type === ApmDocumentType.ServiceTransactionMetric ||
+    type === ApmDocumentType.TransactionMetric
+  ) {
     if (useDurationSummaryField) {
       return TRANSACTION_DURATION_SUMMARY;
     }
```

Comment:
Perhaps we can make this even simpler and ignore the document type. If `useDurationSummaryField = true` we use `TRANSACTION_DURATION_SUMMARY` else `TRANSACTION_DURATION_HISTOGRAM`

```suggestion
    if (useDurationSummaryField) {
      return TRANSACTION_DURATION_SUMMARY;
    }
```